application/src/main/java/com/sanction/thunder/resources/UserResource.java

```diff
@@ -287,4 +302,16 @@ private Response buildResponseForDatabaseError(DatabaseError error, String email
     }
   }
 
+  /**
+   * Determines if the given email string is valid or not.
+   *
+   * @param email The email address to validate.
+   * @return True if the email is valid, false otherwise.
+   */
+  private boolean isValidEmail(String email) {
+    Pattern pattern = Pattern.compile("^.+@.+\\..+$");
```

Comment:
Is the reason you can't catch all bad email cases because this regex doesn't support it or something else?

Reply:
Yeah, this regex doesn't fully capture it. I did some research and apparently if you want to follow the RFC standard there are a ton of weird rules you have to follow. There are some libraries that have email validation classes, but I felt like bringing in a whole library just for this validation wouldn't be a great idea.

This regex is simple and it catches most common bad emails. If we want then we can either try a more complicated regex or bring in one of those libraries.

Reply:
Ok, I think we should use [commons-validator](https://mvnrepository.com/artifact/commons-validator/commons-validator). It has an `EmailValidator` class that should be more than sufficient.

I'll update this PR with that.